File: leetcode/145/Solution.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution:
    def postorderTraversal(self, root: TreeNode) -> List[int]:
        result = []
        if root:
            self.dfs(root, result)
        return result
    
    def dfs(self, root: TreeNode, result: List[int]):
        if root.left:
            self.dfs(root.left, result)
        if root.right:
            self.dfs(root.right, result)
        result.append(root.val)

File: leetcode/145/Solution.java
import java.util.ArrayList;
import java.util.List;

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public List<Integer> postorderTraversal(TreeNode root) {
        List<Integer> result = new ArrayList<>();
        if(root != null) result = dfs(root, result);
        return result;
    }

    public List<Integer> dfs(TreeNode root, List<Integer> result) {
        if(root.left != null) result = dfs(root.left, result);
        if(root.right != null) result = dfs(root.right, result);
        result.add(root.val);
        return result;
    }
}